NotificationServiceExtension/NotificationService.swift

```diff
@@ -0,0 +1,51 @@
+
+import UserNotifications
+import WMF
+
+class FakeLangProvider: WMFPreferredLanguageInfoProvider {
+    func getPreferredContentLanguageCodes(_ completion: @escaping ([String]) -> Void) {
+        completion(["en"])
+    }
+
+    func getPreferredLanguageCodes(_ completion: @escaping ([String]) -> Void) {
+        completion(["en"])
+    }
+
+
+}
+
+class NotificationService: UNNotificationServiceExtension {
+
+    var contentHandler: ((UNNotificationContent) -> Void)?
+    var bestAttemptContent: UNMutableNotificationContent?
+    lazy var remoteNotificationsController: RemoteNotificationsController = {
+        let configuration = Configuration.current
+        let session = Session(configuration: configuration)
+        let fakeProvider = FakeLangProvider()
+        let controller = RemoteNotificationsController(session: session, configuration: configuration, preferredLanguageCodesProvider: fakeProvider)
+        return controller
+    }()
+
+    override func didReceive(_ request: UNNotificationRequest, withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void) {
+        self.contentHandler = contentHandler
+        bestAttemptContent = (request.content.mutableCopy() as? UNMutableNotificationContent)
+        
+        if let bestAttemptContent = bestAttemptContent {
+            // Modify the notification content here...
+
+            remoteNotificationsController.fetchFirstPageNotifications {
```

Comment:
This is just for testing, right? It won't be here in the final version? If so I have a mild preference for a crystal-clear comment here, so we don't accidentally keep it around.

Reply:
Yep, just for testing, but good idea on the comment. I added a comment explaining. I also wouldn't mind having a "Tackle any straggler ToDos" Phab task, I'll make one tomorrow.

Reply:
Thanks!